js/main.js

```diff
@@ -4,51 +4,26 @@ const { combineReducers, applyMiddleware, bindActionCreators } = require('redux'
 const { Provider } = require('react-redux');
 const configureStore = require('./create-store');
 const reducers = require('./reducers');
+const { connectToClient } = require('./thread-client');
 const actions = require('./actions');
-const App = require('./components/app');
 const dom = React.DOM;
 
-const { DebuggerClient } = require('devtools/shared/client/main');
-const { DebuggerTransport } = require('devtools/transport/transport');
-const { TargetFactory } = require("devtools/client/framework/target");
-const socket = new WebSocket("ws://localhost:9000");
-const transport = new DebuggerTransport(socket);
-const client = new DebuggerClient(transport);
+const Toolbox = React.createFactory(require('./components/Toolbox'));
 
-client.connect().then(() => {
-  return client.listTabs().then(response => {
-    const tab = response.tabs[response.selected];
-    const options = { form: tab, client, chrome: false };
-    TargetFactory.forRemoteTab(options).then(target => {
-      target.activeTab.attachThread({}, (res, threadClient) => {
-        threadClient.resume();
-        window.gThreadClient = threadClient;
-        connectThread(gThreadClient);
-      });
-    });
-  });
-}).catch(err => console.log(err));
 
 const createStore = configureStore({ log: true });
 const store = createStore(combineReducers(reducers));
+window.store = store;
 
-function connectThread(threadClient) {
-  threadClient.addListener("newSource", (event, packet) => {
-    store.dispatch(actions.newSource(packet.source));
-  });
-
-  threadClient.addListener("paused", (_, packet) => {
-    console.log(packet);
-  });
-
-  store.dispatch(actions.loadSources())
-}
```

Comment:
It's so nice to see red in a bootstrap file like main.js. With that said, I tried to not lock us into a solution with too rigid an API.